### PR TITLE
Bug: fsx windows fileserver SSM arn parsing was incorrect

### PR DESCRIPTION
**Pull request by arun-annamalai**
Currently the parsing of the ssm parameter name in the fsx windows file server workflow is incorrect. This is causing customers incorrect behavior when using ssm parameters that have a folder hierarchy.

### Summary

This PR is to fix the above bug and allow customers to use ssm parameters that have a folder hierarchy.

### Testing

Unit Tests

New tests cover the changes:  yes

### Description for the changelog

Bug: fsx windows fileserver SSM arn parsing was incorrect

**Does this PR include breaking model changes? If so, Have you added transformation functions?**
  
No model changes
### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows_test.go [hunk 1]
```diff
@@ -167,35 +167,64 @@ func TestRetrieveCredentials(t *testing.T) {
...
+	cases := []struct {
+		Name                     string
+		CredentialsParameterARN  string
+		CredentialsParameterName string
+	}{
+		{
+			Name:                     "TestRetrieveSSMCredentialsSimple",
+			CredentialsParameterARN:  "arn:aws:ssm:us-west-2:123456789012:parameter/test",
+			CredentialsParameterName: "/test",
+		},
```
Comment (yinyic) [inline agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows_test.go:179]: Is this test case redundant?

Comment (arun-annamalai) [reply agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows_test.go:179]: yes it is, let me get rid of it.

**Review by yinyic — APPROVED** [review]

**Comment by arun-annamalai** [comment]
Will be posting integ test results here when completed.

File: agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go [hunk 2]
```diff
@@ -479,7 +478,9 @@ func (fv *FSxWindowsFileServerResource) retrieveSSMCredentials(credentialsParame
 	}
 
 	ssmClient := fv.ssmClientCreator.NewSSMClient(fv.region, iamCredentials)
-	ssmParam := filepath.Base(parsedARN.Resource)
+	// parsedARN.Resource looks like "arn:aws:ssm:us-west-2:123456789012:parameter/sample1/sample2/parameter1"
+	// We split by parameter and get ["", "/sample1/sample2/parameter1"]
+	ssmParam := strings.Split(parsedARN.Resource, "parameter")[1]
```
Comment (amogh09) [inline agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go:483]: `parsedARN.Resource` is guaranteed to have "parameter" in it?

**Comment by jiuchoe4** [comment]
Integ test results: successfully reproduced the issue wherein task with FSx using SSM parameter for credentials fails with `Invalid parameters: fsx`when the parameter has a path. Also verified that using a parameter without a path worked successfully. Pulled and built this commit, then initialized the new agent on the instance and reran the task with a successful run

File: agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go (hunk 2 above)
Comment (arun-annamalai) [reply agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go:483]: yes! if you look at the caller of this code, we use the Amazon ARN library to parse the string and make sure its a SSM parameter. This avoids any injections and other security issues.

Comment (amogh09) [reply agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go:483]: Thanks!

**Review by amogh09 — APPROVED** [review]

File: agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go [hunk 3]
```diff
@@ -479,15 +478,24 @@ func (fv *FSxWindowsFileServerResource) retrieveSSMCredentials(credentialsParame
 	}
 
 	ssmClient := fv.ssmClientCreator.NewSSMClient(fv.region, iamCredentials)
-	ssmParam := filepath.Base(parsedARN.Resource)
-	ssmParams := []string{ssmParam}
+	// parsedARN.Resource looks like "arn:aws:ssm:us-west-2:123456789012:parameter/sample1/sample2/parameter1"
+	// We cut by parameter and get "arn:aws:ssm:us-west-2:123456789012:parameter", "/sample1/sample2/parameter1", True/False
+	_, ssmParamName, found := strings.Cut(parsedARN.Resource, "parameter")
+	if !found {
+		err = errors.New("unxpected error. expected fsx credential ssm arn but did not find string 'parameter' in the arn")
```
Comment (amogh09) [inline agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go:485]: How about adding `parsedARN.Resource` to the error string?

Comment (arun-annamalai) [reply agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go:485]: I was hesitant for security reasons, there were some concerns around doing that when we implemented domainless gMSA

Comment (amogh09) [reply agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go:485]: oh okay

**Review by amogh09 — APPROVED** [review]

**Review by yinyic — APPROVED** [review]

**Pull request merged**